articles/coding-on-purpose.md

```diff
@@ -0,0 +1,99 @@
+
+> Make something people want.
+
+By the way: Justin Bieber - born 1994 - is a millenial too. And so it's no surprise that his current (2017) world tour is titled the *Purpose World Tour*.
+
+When building software, the *why* or better the *what for* question is a very important one. I'm aware that this is nothing new for you. But this is what makes it easy to forget and to lose the focus on it. We usually start our projects with a more or less clear vision of what we want to achieve - how the world will be better when we're done. Then we start building. We start setting milestones, and soon reaching the milestone becomes our highest goal. In order to meet milestones we need to deliver a certain amount of user stories per sprint (or so we think...) and soon estimating effort in story points and tracking velocity become our way of steering the project. Or we might be busy optimizing the hourly rate of our sub-contractors from another continent. Whenever our focus drifts away from achieving the original purpose, we need to remind ourselves:
+
+> When we build software, our success is measured by whether the software can make the impact for which the customer is ready paying a certain amount of money.
+
+You don't need to be a millenial to understand this. This was true long before millenials were even born.
+
+## Purpose in Software Development Methodologies
+
+As long as there has been a professional software industry people in this industry have invented methods for approaching software development. There's one theme that shows up again and again in all methods: *managing purpose*.
+
+* In *RUP*, one tries to establish traceability from lower level items to higher level goals
+* If you work with *Use Cases* each Use Case represents a goal on a certain abstraction level and you can show how lower level Use Cases support higher level Use Cases in a use case diagram.
+* *User Stories* are usually formulated according to a template that also includes a *goal* and a *benefit* (e.t. "as a &lt;role&gt; I want to &lt;goal&gt; so that &lt;benefit&gt;").
+* With a *User Story Map* we can create a hierarchy of these *User Stories*.
+* An *Impact Map* traces down the high level goals to actors, impacts and finally deliverables
+* *Design Thinking* has a strong focus on *identifying and understanding customer needs*
+* The idea of the *Minimal Viable Product* makes the point that we should only build as much as is necessary for fulfilling the desired goal.
+
+And even beyond software develpment, the *why* topic seems to be very popular. One of the most famous representatives of the modern *why* movement is probably [Simon Sinek who tells us](https://startwithwhy.com/): *Start with Why!*
+
+## A Vacuum Of Purpose
+
+Let's go back to software again. Even though there's a lot of talk about impact, purpose and so on, there are also many areas with a vacuum of purpose. Some examples:

```

Comment:
Warum "let's go back"? Hat sich der Text von Software (engineering) wegentwickelt? Ich meine nicht.

Ich würde den ersten Satz weglassen. Oder "lets dive into some software engineering disciplines", irgendwie so.

Reply:
Keep it short here "Unfortunately there are many areas that have no direct conjunction to the Purpose - some examples:"

Reply:
I would really bring this psart here first after the introduction - this would motivate to find out later which techniques are more related to a Purpose.